packages/legend-query/src/components/QueryBuilderValueSpecificationEditor.tsx

```diff
@@ -487,6 +495,76 @@ export const LatestDatePrimitiveInstanceValueEditor: React.FC = () => (
   </div>
 );
 
+export const DateInstanceValueEditor = observer(
+  (props: {
+    valueSpecification: PrimitiveInstanceValue;
+    graph: PureModel;
+    expectedType: Type;
+    className?: string | undefined;
+  }) => {
+    const { valueSpecification, graph, expectedType, className } = props;
+    const variableType = valueSpecification.genericType.value.rawType;
+    const selectedType = buildElementOption(variableType);
+    const typeOptions: PackageableElementOption<Type>[] = graph.primitiveTypes
+      .filter(
+        (p) =>
+          p.name === PRIMITIVE_TYPE.STRICTDATE ||
+          p.name === PRIMITIVE_TYPE.DATETIME ||
+          p.name === PRIMITIVE_TYPE.LATESTDATE,
+      )
+      .map((p) => buildElementOption(p) as PackageableElementOption<Type>);
+
+    const changeType = (val: PackageableElementOption<Type>): void => {
+      if (variableType !== val.value) {
+        valueSpecification.genericType.value.setRawType(val.value);
+      }
+      if (
+        valueSpecification.genericType.value.rawType.name !==
+        PRIMITIVE_TYPE.LATESTDATE
+      ) {
+        valueSpecification.values = [
+          format(
+            new Randomizer().getRandomDate(
+              new Date(Date.now()),
+              addDays(Date.now(), 100),
+            ),
+            DATE_FORMAT,
+          ),
+        ];
+      }
+    };
+
+    return (
+      <div className="query-builder-value-spec-editor__date">
+        {(valueSpecification.genericType.value.rawType.name ===
```

Comment:
could we instead of comparing names compares the types themselves. You can grab
````suggestion
const strictDtae = graph.getPrimitiveType(PRIMITETYPE.STRICTDATE);
rawType ===strictDate
```